Rubberduck.Parsing/Symbols/ParsingExceptions/RubberduckParseErrorListenerBase.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using Antlr4.Runtime;
+using Rubberduck.Parsing.VBA;
+
+namespace Rubberduck.Parsing.Symbols.ParsingExceptions
+{
+    public class RubberduckParseErrorListenerBase : BaseErrorListener, IRubberduckParseErrorListener
+    {
+        public RubberduckParseErrorListenerBase(CodeKind codeKind)
+        {
+            CodeKind = codeKind;
+        }
+
+        protected CodeKind CodeKind { get; }
+        
+        //This serves as a method to postpone throwing a parse exception to after the entire input has been parsed,
+        //e.g. when recovering from errors and collecting them.
+        public virtual bool HasPostponedException(out Exception exception)
```

Comment:
No, the idea was that you could define an exception with a collection of parserErrors as payload. You would generate that at the first exception an provide it with the reference to the collection of errors on your listener. Then you simply keep adding to the list of errors. After the parse has finished, the parser throws the exception.

Reply:
Would it be better to return a specialized custom exception class than just `Exception` then?

Reply:
Why should we limit the exception type new error listeners return as postponed exception?

Reply:
Primarily because if we intend to make the parser error(s) the payload of the exception, it's good to have an explicit API for examining the payload, no? If they aren't parser errors, we have an entirely different problem anyway. Besides, creating a custom exception doesn't limit you to only that one; derivatives can be made if we need to process different kind of parser errors. Besides, that would offer us the opportunity to encapsulate the creating and adding of the parser errors to the exception that will be eventually returned.

Reply:
The idea was that the specific exception type will be caught at the point where the errors should be examined. Anyway, I would like to leave  any of the implementation of the exception and the handling to a future PR of someone actually interested to implement error recovery. I just added the setup here because it was argued before that the parsing setup needed to be completely rewritten in order to allow error recovery.